app/src/main/java/org/mozilla/tv/firefox/settings/SettingsFragment.kt

```diff
@@ -55,14 +55,20 @@ class SettingsFragment : Fragment() {
             builder1.setPositiveButton(
                 getString(R.string.action_ok)
             ) { dialog, _ -> with(requireContext()) {
-                    requireWebRenderComponents.engine.deleteData(this)
-                    requireWebRenderComponents.sessionManager.removeAll()
+                    val components = requireWebRenderComponents
+                    components.engine.deleteData(this)
+                    components.sessionManager.removeAll()
                     dialog.cancel()
+                    serviceLocator.webViewCache.doNotPersist(doAfterRecreate = {
+                        // Sessions must be removed prior to MainActivity being recreated, in order for initialization
+                        // logic to be called. However, the URL is set before the new WebView is created, causing it to
```

Comment:
"The URL is set before the new WebView is created, causing it be out of sync with the session."

I don't understand, can you elaborate? e.g. Where is the URL set? Who sets it? What causes it to be out of sync?

Reply:
I want to provide context into why this happens, but I'm also concerned about going into so much detail that it will more quickly bitrot.

I'll see if I can strike a better balance on this.